swap_meet/vendor.py

```diff
@@ -1,2 +1,66 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+    
+    def get_by_category(self, category, ):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+    
+    def swap_items(self, vendor_b, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor_b.inventory:
+            return False
+
+        self.inventory.append(their_item)
+        vendor_b.inventory.append(my_item)
+        self.inventory.remove(my_item)
+        vendor_b.inventory.remove(their_item)
+        return True
+
+    def swap_first_item(self, vendor_b):
+        
+        if not self.inventory or not vendor_b.inventory:
+            return False
+            
+        swapped_items = self.swap_items(vendor_b, self.inventory[0], vendor_b.inventory[0])
+            
+        return swapped_items
+
+    def get_best_by_category(self, category):
+        items_match_category = self.get_by_category(category)
+        
+        if not items_match_category:
+            return None
+
+        init_condition= 0
+        for item in items_match_category:
+            if item.condition > init_condition:
+                init_condition = item.condition
+                best_item = item
+        return best_item
```

Comment:
You can access variables defined in for/if loops outside of those said for/if loops (like here on line 57 where you return `best_item` when it was declared in an if block) because Python’s smallest scope is a function and not a block.

However, this can be confusing and I think it's more explicit to declare best_item outside of your for loop, like on line 51. 

If you had `best_item = None` on line 51 then you can remove line 49 and 50. If items_match_category is empty then the for loop on line 53 doesn't do anything, and you'll just return best_item (which is already None).